### PR TITLE
Introduce Boltalyzer, a Bolt PCAP analysis tool

### PR DESCRIPTION
**Pull request by jakewins**
- Allows parsing a `pcap` file from eg. `tcpdump` and output a
  human-readable description of the network traffic. This can help
  understanding which side of an exchange is performing poorly - eg. where
  latency is accumulated.
- Break `tools` module into submodules for different tools
- Refactor out the dechunking code from BoltProtocolV1 to reuse
  in Boltalyzer
## Example;

```
$ java -jar boltalyzer.jar capture.pcap

00000003 session-000 Client <HANDSHAKE>
00000003 session-000 Server <EMPTY>
00000012 session-000 Server <HANDSHAKE RESPONSE>
00000013 session-000 Client <EMPTY>
00000034 session-000 Client
  INIT
  RUN CREATE (n {name:{name}}) RETURN id(n)
  PULL_ALL
00000034 session-000 Server <EMPTY>
00000244 session-000 Server
  SUCCESS
  SUCCESS
  RECORD
  SUCCESS
```

### CODE REVIEW

File: tools/traffic-analyzer/src/main/java/org/neo4j/tools/boltalyzer/PCAPParser.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.neo4j.function.ThrowingSupplier;
+
+
+public class PCAPParser
+{
+    private final int PCAP_HEADER = 0xA1B2C3D4;
```
Comment (pontusmelke) [inline tools/traffic-analyzer/src/main/java/org/neo4j/tools/boltalyzer/PCAPParser.java:41]: static?

File: tools/traffic-analyzer/src/main/java/org/neo4j/tools/boltalyzer/PCAPParser.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    public static final int IP_VERSION_AND_LENGTH_OFFSET = 0;
+    public static final int IP_SRC_OFFSET = 12;
+    public static final int IP_DST_OFFSET = 16;
+
+    public static final int TCP_SRC_PORT_OFFSET = 0;
+    public static final int TCP_DST_PORT_OFFSET = 2;
+
+
+    private static final Map<Integer,PhysicalFormat> physicalFormats = new HashMap<>();
+    private static byte[] IP_BUFFER = new byte[4];
```
Comment (pontusmelke) [inline tools/traffic-analyzer/src/main/java/org/neo4j/tools/boltalyzer/PCAPParser.java:52]: final?

File: tools/traffic-analyzer/pom.xml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    <version>3.0.0-SNAPSHOT</version>
+    <relativePath>../..</relativePath>
+  </parent>
+
+  <artifactId>neo4j-traffic-analyzer</artifactId>
+  <version>3.0.0-SNAPSHOT</version>
+
+  <name>Neo4j - Bolt Traffic Analyzer</name>
+  <packaging>jar</packaging>
+  <description>Analyzes binary traffic dumps for the Bolt protocl, allowing performance analysis et cetera</description>
```
Comment (pontusmelke) [inline tools/traffic-analyzer/pom.xml:19]: protocl -> protocol

**Comment by jakewins** [comment]
@pontusmelke fixed your comments and some others from @lutovich, openened new PR here; https://github.com/neo4j/neo4j/pull/5963

**Pull request closed without merging**